### PR TITLE
fix(osv): skip withdrawn entries

### PR DESCRIPTION
**Pull request by nikpivkin**
## Description
See https://github.com/aquasecurity/trivy/issues/4621

### CODE REVIEW

File: pkg/vulnsrc/osv/osv.go [hunk 1]
```diff
@@ -123,6 +123,11 @@ func (vs VulnSrc) save(eco ecosystem, entries []Entry) error {
 }
 
 func (vs VulnSrc) commit(tx *bolt.Tx, eco ecosystem, entry Entry) error {
+
+	if entry.Withdrawn != nil {
```
Comment (afdesk) [inline pkg/vulnsrc/osv/osv.go:127]: maybe len() will be better?
```suggestion
	if len(entry.Withdrawn) > 0 {
```

Comment (nikpivkin) [reply pkg/vulnsrc/osv/osv.go:127]: `Withdrawn` is a pointer to time.Time

Comment (afdesk) [reply pkg/vulnsrc/osv/osv.go:127]: oh, ok.

Comment (knqyf263) [reply pkg/vulnsrc/osv/osv.go:127]: Do we want to compare the date? The following code doesn't work probably, but I just wanted to explain my thought.

```suggestion
	if entry.Withdrawn != nil && entry.Withdrawn.Before(time.Now()){
```

File: pkg/vulnsrc/osv/testdata/happy/vuln-list/osv/python/withdrawn_after_now.json [hunk 2]
```diff
@@ -0,0 +1,102 @@
+{
+  "id": "PYSEC-2019-156",
+  "modified": "2021-07-05T00:01:27.093286Z",
+  "published": "2019-11-04T20:15:00Z",
+  "withdrawn": "2024-06-06T10:37:00Z",
```
Comment (knqyf263) [inline pkg/vulnsrc/osv/testdata/happy/vuln-list/osv/python/withdrawn_after_now.json:5]: This test will fail next year. Can we change it 3000 or something?

**Review by knqyf263 — APPROVED** [review]

File: pkg/vulnsrc/osv/testdata/happy/vuln-list/osv/python/withdrawn_after_now.json (hunk 2 above)
Comment (afdesk) [reply pkg/vulnsrc/osv/testdata/happy/vuln-list/osv/python/withdrawn_after_now.json]: let's wait for it.
I'm going to see it. while everything is going well ))

**Pull request merged**